python/ray/serve/_private/proxy_request_response.py

```diff
@@ -160,28 +161,30 @@ def user_request(self) -> bytes:
     def send_request_id(self, request_id: str):
         self.context.set_trailing_metadata([("request_id", request_id)])
 
-    def send_status_code(self, status_code: grpc.StatusCode):
-        self.context.set_code(status_code)
-
-    def send_details(self, message: str):
-        self.context.set_details(message)
-
     def request_object(self, proxy_handle: ActorHandle) -> gRPCRequest:
         return gRPCRequest(
             grpc_user_request=self.user_request,
             grpc_proxy_handle=proxy_handle,
         )
 
 
-class ProxyResponse:
-    """ProxyResponse class to use in the common interface among proxies"""
+@dataclass
+class ResponseStatus:
+    code: Any  # Must be convertible to a string.
+    is_error: bool = False
+    message: str = ""
 
-    def __init__(
-        self,
-        status_code: str,
-        response: Optional[bytes] = None,
-        streaming_response: Optional[Generator[bytes, None, None]] = None,
-    ):
-        self.status_code = status_code
-        self.response = response
-        self.streaming_response = streaming_response
+
+# Yields protocol-specific messages followed by a final `ResponseStatus`.
+ResponseGenerator = AsyncIterator[Union[Any, ResponseStatus]]
```

Comment:
:)   why do we have `Any` in the Union type

Reply:
Bc that message can be any type (e.g. protobuf, serialized json, dict, int...etc anything the deployment returns) and is shared between HTTP and gRPC calls

Reply:
why can't  `ResponseStatus` be one of them?

Reply:
Oh you meant just do `AsyncIterator[Any]`? I feel it's better to explicitly state `ResponseStatus`. Would prefer to keep as is if not strong feeling to it

Reply:
This is here to make it more clear what the sequence of messages is: N `Any` followed by one final `ResponseStatus`. It's essentially just documentation.